docs/dev/metrics.md

```diff
@@ -0,0 +1,108 @@
+# Metrics Naming
+
+## TL;DR
+
+**Metric Naming Guide**: Prioritize Developer Experience (DX) with intuitive, readable names that follow consistent conventions. Start with a domain prefix, use snake_case, avoid plurals in names (except counters), include units where relevant, and use labels for variations. Align with Prometheus and OpenTelemetry standards, and adhere to UCUM for units.
+
+## Definitions
+
+- Metric: is a measurement used to track the state and behavior of a system component. Metrics represent the current status (e.g., readiness or availability) or describe specific operations (e.g., request counts, cache hits, latencies, failures) to provide insight into system health, performance, and workload.
+- Metric Name: The identifier for the metric, typically composed of a domain or namespace prefix followed by the metric's specific name.
+  - Metric domain or namespace:  refers to a logical grouping or category that organizes related metrics. It typically indicates the component, system, or protocol that the metric belongs to and helps avoid naming conflicts and provides context for interpreting the metric.
+- Label (or Dimension): Key-value pairs that provide additional context to a metric.
+- Counter-Based Metrics: Metrics that represent a cumulative count of events or occurrences.
+- UpDownCounter-Based Metrics: Counter-Based Metrics that can increase or decrease over time, unlike regular counters which only increase. They are used for values that fluctuate, such as current active connections or in-progress operations.
```

Comment:
```suggestion
- Gauge-Based Metrics: Counter-Based Metrics that can increase or decrease over time, unlike regular counters which only increase. They are used for values that fluctuate, such as current active connections or in-progress operations.
```

Gauge is the OTel terminology for this, which I think more accurately describes it: https://opentelemetry.io/docs/specs/otel/metrics/data-model/#gauge

Reply:
Updated, thank you!